To: vim_dev@googlegroups.com
Subject: Patch 7.3.061
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.061
Problem:    Remote ":drop" does not respect 'autochdir'. (Peter Odding)
Solution:   Don't restore the directory when 'autochdir' is set. (Benjamin
	    Fritz)
Files:	    src/main.c


*** ../vim-7.3.060/src/main.c	2010-09-29 17:26:57.000000000 +0200
--- src/main.c	2010-11-16 16:16:11.000000000 +0100
***************
*** 3817,3822 ****
--- 3817,3824 ----
      /* Check if we have at least one argument. */
      if (filec <= 0)
  	mainerr_arg_missing((char_u *)filev[-1]);
+ 
+     /* Temporarily cd to the current directory to handle relative file names. */
      if (mch_dirname(cwd, MAXPATHL) != OK)
  	return NULL;
      if ((p = vim_strsave_escaped_ext(cwd,
***************
*** 3858,3870 ****
  	ga_concat(&ga, p);
  	vim_free(p);
      }
      /* The :drop commands goes to Insert mode when 'insertmode' is set, use
       * CTRL-\ CTRL-N again. */
!     ga_concat(&ga, (char_u *)"|if exists('*inputrestore')|call inputrestore()|endif<CR>");
!     ga_concat(&ga, (char_u *)"<C-\\><C-N>:cd -");
      if (sendReply)
! 	ga_concat(&ga, (char_u *)"<CR>:call SetupRemoteReplies()");
!     ga_concat(&ga, (char_u *)"<CR>:");
      if (inicmd != NULL)
      {
  	/* Can't use <CR> after "inicmd", because an "startinsert" would cause
--- 3860,3879 ----
  	ga_concat(&ga, p);
  	vim_free(p);
      }
+     ga_concat(&ga, (char_u *)"|if exists('*inputrestore')|call inputrestore()|endif<CR>");
+ 
      /* The :drop commands goes to Insert mode when 'insertmode' is set, use
       * CTRL-\ CTRL-N again. */
!     ga_concat(&ga, (char_u *)"<C-\\><C-N>");
! 
!     /* Switch back to the correct current directory (prior to temporary path
!      * switch) unless 'autochdir' is set, in which case it will already be
!      * correct after the :drop command. */
!     ga_concat(&ga, (char_u *)":if !exists('+acd')||!&acd|cd -|endif<CR>");
! 
      if (sendReply)
! 	ga_concat(&ga, (char_u *)":call SetupRemoteReplies()<CR>");
!     ga_concat(&ga, (char_u *)":");
      if (inicmd != NULL)
      {
  	/* Can't use <CR> after "inicmd", because an "startinsert" would cause
*** ../vim-7.3.060/src/version.c	2010-11-16 15:48:57.000000000 +0100
--- src/version.c	2010-11-16 16:19:58.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     61,
  /**/

-- 
BEDEVERE: Stand by for attack!!
   [CUT TO enormous army forming up.  Trebuchets, rows of PIKEMEN, siege
   towers, pennants flying, shouts of "Stand by for attack!"  Traditional
   army build-up shots.  The shouts echo across the ranks of the army.
   We see various groups reacting, and stirring themselves in readiness.]
ARTHUR:   Who are they?
BEDEVERE: Oh, just some friends!
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
